util/src/linear_map.rs

```diff
@@ -1,11 +1,22 @@
 use alloc::vec::Vec;
 use core::mem;
 
-/// O(n) Vec-backed map for keys that only implement Eq.
-/// Only use this for a very small number of keys, operations
-/// on it can easily become O(n^2).
+/// A linear key-value map backed by a `Vec`.
+///
+/// This map performs **O(n)** lookups and inserts.
+/// It is suitable only for **small** sets of keys where
+/// performance is not critical and keys implement `Eq` but not `Hash`.
```

Comment:
I don't think anything goes wrong if the Key's implement Hash. (You just `could` use a hashmap in that case if you wanted to). I also think the performance of this should be pretty good, provided the number of keys is very small (In our examples it's `1` or `2`). It's probably even faster than a hashmap for such small values and should be very efficient space wise.

```suggestion
/// This map performs **O(n)** lookups and inserts.
/// It is suitable only for **small** sets of keys which
/// must implement `Eq`.
```

Reply:
Ok understood I thought the use case was for elements that don't implement `Hash`. Fixed